### ISSUE TITLE
Check passes even if it failed

### ISSUE THREAD

**Issue by VincentGuyader** [r-lib/rcmdcheck#56]
HI,

I create a function to batch test somes student package : 
```

#' run check on github
#' 
#' @param git_url
#' @export
#' @import git2r
#' @import rcmdcheck
#' @import glue
#'
ca_passe_le_ckeck <- function(git_url){
  doss <- file.path(tempdir(),git_url,abs(rnorm(1)))
  dir.create(doss)
  git2r::clone(glue('https://github.com/{git_url}.git'),doss)
  rcmdcheck::rcmdcheck(doss)
}
```
But even if error and warning are detected the output is "0 errors ✔ | 0 warnings ✔ | 0 notes ✔" :

``` r
library(glue)
library(git2r)
library(rcmdcheck)
ca_passe_le_ckeck <- function(git_url){
  doss <- file.path(tempdir(),git_url,abs(rnorm(1)))
  dir.create(doss)
  git2r::clone(glue('https://github.com/{git_url}.git'),doss)
  rcmdcheck::rcmdcheck(doss)
}

ca_passe_le_ckeck(git_url = "mariafcadena/HomeworkCADENA")
```

<details><summary>full output</summary>
  
  ``` r


ca_passe_le_ckeck(git_url = "mariafcadena/HomeworkCADENA")
cloning into 'C:\Users\vincent\AppData\Local\Temp\RtmpKatUa2/mariafcadena/HomeworkCADENA/0.333691129026097'...
Receiving objects:   1% (1/62),   32 kb
Receiving objects:  11% (7/62),   32 kb
Receiving objects:  22% (14/62),   32 kb
Receiving objects:  32% (20/62),   32 kb
Receiving objects:  41% (26/62),   32 kb
Receiving objects:  51% (32/62),   32 kb
Receiving objects:  61% (38/62),   32 kb
Receiving objects:  72% (45/62),   32 kb
Receiving objects:  82% (51/62),   32 kb
Receiving objects:  91% (57/62),   32 kb
Receiving objects: 100% (62/62),   32 kb, done.
─  using log directory 'C:/Users/vincent/AppData/Local/Temp/RtmpKatUa2/file378031625d18/MariaCADENA.Rcheck' (584ms)
─  using R version 3.4.2 (2017-09-28)
─  using platform: x86_64-w64-mingw32 (64-bit)
─  using session charset: ISO8859-1
✔  checking for file 'MariaCADENA/DESCRIPTION'
─  checking extension type ... Package
─  this is package 'MariaCADENA' version '0.1.0'
─  package encoding: UTF-8
✔  checking package namespace information ...
✔  checking package dependencies (1.4s)
✔  checking if this is a source package ...
✔  checking if there is a namespace
✔  checking for .dll and .exe files
✔  checking for hidden files and directories ... 
✔  checking for portable file names ... 
✔  checking whether package 'MariaCADENA' can be installed (4.2s)
✔  checking package directory
✔  checking 'build' directory
N  checking DESCRIPTION meta-information (668ms)
   Malformed Description field: should contain one or more complete sentences.
✔  checking top-level files (696ms)
✔  checking for left-over files ...
✔  checking index information (546ms)
✔  checking package subdirectories (705ms)
✔  checking R files for non-ASCII characters ... 
✔  checking R files for syntax errors ... 
✔  checking whether the package can be loaded (1.2s)
✔  checking whether the package can be loaded with stated dependencies (1.2s)
✔  checking whether the package can be unloaded cleanly (1.2s)
✔  checking whether the namespace can be loaded with stated dependencies (1.2s)
✔  checking whether the namespace can be unloaded cleanly (1.3s)
✔  checking loading without being on the library search path (1.7s)
✔  checking dependencies in R code (1.2s)
✔  checking S3 generic/method consistency (1.7s)
✔  checking replacement functions (1.2s)
✔  checking foreign function calls (1.2s)
N  checking R code for possible problems (5.2s)
   draw_a_name: no visible binding for global variable 'sex'
   draw_a_name: no visible binding for global variable 'name'
   draw_a_name: no visible binding for global variable 'year'
   draw_a_name: no visible binding for global variable 'total'
   draw_names: no visible binding for global variable 'name'
   draw_names: no visible binding for global variable 'year'
   draw_names: no visible binding for global variable 'total'
   my_csv_reader: no visible binding for global variable 'read.csv2'
   save_as_csv: no visible global function definition for 'write.csv2'
   Undefined global functions or variables:
     name read.csv2 sex total write.csv2 year
   Consider adding
     importFrom("utils", "read.csv2", "write.csv2")
   to your NAMESPACE file.
N  checking Rd files (448ms)
   prepare_Rd: my_csv_reader.Rd:18-20: Dropping empty section \examples
✔  checking Rd metadata ... 
✔  checking Rd cross-references (356ms)
✔  checking for missing documentation entries (1.3s)
✔  checking for code/documentation mismatches (3.6s)
✔  checking Rd \usage sections (2.1s)
✔  checking Rd contents (349ms)
W  checking for unstated dependencies in examples (361ms)
   Warning: erreur d'analyse de code dans le fichier 'MariaCADENA-Ex.R' :
   9: '{' inattendu(e)
   19: 
   20: dontrun {
               ^
✔  checking installed files from 'inst/doc'
✔  checking files in 'vignettes' ...
E  checking examples (1.7s)
   Running examples in 'MariaCADENA-Ex.R' failed
   The error most likely occurred in:
   
   > ### Name: draw_a_name
   > ### Title: Drwas the timers series of a certain name of a certain sex
   > ### Aliases: draw_a_name
   > 
   > ### ** Examples
   > 
   > dontrun {draw_a_name(c("Marie", "F"))}
   Error: unexpected '{' in "dontrun {"
   Execution halted
✔  checking for unstated dependencies in vignettes (553ms)
✔  checking package vignettes in 'inst/doc' ...
─  checking running R code from vignettes ...
      'MariaCADENA.Rmd' using 'UTF-8' ... OK
    NONE
✔  checking re-building of vignette outputs (2.1s)
─  checking PDF version of manual (4.7s)
   OK
   See
     'C:/Users/vincent/AppData/Local/Temp/RtmpKatUa2/file378031625d18/MariaCADENA.Rcheck/00check.log'
   for details.
   
── R CMD check results ──────────────────────────────────────────────── MariaCADENA 0.1.0 ────
Duration: 46.2s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔



```

</details>

**Comment by VincentGuyader** [comment]
It works on unbutu, not on my windows computer : 
```
> sessionInfo()
R version 3.4.2 (2017-09-28)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=French_France.1252      LC_CTYPE=French_France.1252       
[3] LC_MONETARY=French_France.1252     LC_NUMERIC=C                      
[5] LC_TIME=English_United States.1252

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_1.0.5 glue_1.2.0 

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.13         rstudioapi_0.7       whisker_0.3-2        knitr_1.17          
 [5] magrittr_1.5         rcmdcheck_1.2.1.9003 devtools_1.13.4      debugme_1.1.0       
 [9] xtable_1.8-2         R6_2.2.2             stringr_1.2.0        tools_3.4.2         
[13] miniUI_0.1.1         clipr_0.3.3          git2r_0.19.0         withr_2.1.0.9000    
[17] shinyjs_0.9.1        htmltools_0.3.6      yaml_2.1.14          assertthat_0.2.0    
[21] rprojroot_1.2        digest_0.6.12        crayon_1.3.4         processx_2.0.0.1    
[25] callr_1.0.0.9000     clisymbols_1.2.0     evaluate_0.10.1      memoise_1.1.0       
[29] mime_0.5             rmarkdown_1.6        reprex_0.1.1         stringi_1.1.5       
[33] compiler_3.4.2       desc_1.1.1           backports_1.1.1      prettyunits_1.0.2   
[37] jsonlite_1.5         httpuv_1.3.5  
```

**Comment by gaborcsardi** [comment]
Where is this package?

```
Package required but not available: ‘prenoms’

See section ‘The DESCRIPTION file’ in the ‘Writing R Extensions’
manual.
* DONE
```

**Comment by gaborcsardi** [comment]
OK, I can reproduce this, fixing it now.

**Comment by gaborcsardi** [comment]
Should be fixed in 22d14f7b3de2c9cf319f6cbc777d728def5c646b, please try.

**Comment by VincentGuyader** [comment]
same issue with 1.2.1.9004

Do you need something to help ?

**Comment by gaborcsardi** [comment]
Are you sure you tried it in a new R session?

**Comment by VincentGuyader** [comment]
Hi gabor, 

yes, new R session.
and I tried again today after a reboot.

Regards

**Comment by gaborcsardi** [comment]
Right. Please try again.

**Comment by VincentGuyader** [comment]
🥇
